ambry-router/src/test/java/com/github/ambry/router/NonBlockingRouterQuotaCallbackTest.java

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.github.ambry.commons.RetainingAsyncWritableChannel;
+import com.github.ambry.messageformat.MessageFormatRecord;
+import com.github.ambry.protocol.GetOption;
+import com.github.ambry.quota.QuotaChargeCallback;
+import com.github.ambry.quota.QuotaMode;
+import com.github.ambry.utils.Utils;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.utils.TestUtils.*;
+import static org.junit.Assert.*;
+
+
+/**
+ * Class to test the {@link NonBlockingRouter} with quota callbacks.
+ */
+public class NonBlockingRouterQuotaCallbackTest extends NonBlockingRouterTestBase {
```

Comment:
I think taking QuotaCallback test out from NonBlockingRouterTest is unnecessary. We only have one test method and we are take huge chunk of code and create a base file. It's too many changes for too little gain.

Reply:
The main motivation behind taking this out was that I want to test it for a much large number of parameters. And I don't want to test all the test cases of original NonBlockingRouterTest for all those parameters. This slows down our test time for the build.